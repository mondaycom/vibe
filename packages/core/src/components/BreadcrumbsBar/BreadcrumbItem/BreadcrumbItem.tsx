import { ComponentDefaultTestId, getTestId } from "../../../tests/test-ids-utils";
import cx from "classnames";
import React, { useRef } from "react";
import useIsOverflowing from "../../../hooks/useIsOverflowing/useIsOverflowing";
import Tooltip from "../../../components/Tooltip/Tooltip";
import { BreadcrumbContent } from "./BreadcrumbContent/BreadcrumbContent";
import { SubIcon, VibeComponentProps } from "../../../types";
import styles from "./BreadcrumbItem.module.scss";

export interface BreadcrumbItemProps extends VibeComponentProps {
  /**
   * The display text of the breadcrumb item.
   */
  text?: string;
  /**
   * If true, the item is disabled.
   */
  disabled?: boolean;
  /**
   * If true, the item is clickable.
   */
  isClickable?: boolean;
  /**
   * The URL the item links to if navigation is handled via a link.
   */
  link?: string;
  /**
   * Callback fired when the item is clicked.
   */
  onClick?: () => void;
  /**
   * If true, applies styling for the current page.
   */
  isCurrent?: boolean;
  /**
   * The icon displayed next to the text.
   */
  icon?: SubIcon;
  /**
   * If true, the breadcrumb text is shown.
   */
  showText?: boolean;
}

const BreadcrumbItem: React.FC<BreadcrumbItemProps> = ({
  className,
  text = "",
  disabled,
  isClickable = false,
  link,
  onClick,
  isCurrent = false,
  icon,
  id,
  showText = true,
  "data-testid": dataTestId
}) => {
  const componentRef = useRef<HTMLSpanElement>(null);
  const isOverflowing = useIsOverflowing({ ref: componentRef });

  return (
    <Tooltip
      disableDialogSlide={true}
      withoutDialog={false}
      content={(isOverflowing || !showText) && text}
      showTrigger={["mouseenter"]}
      hideTrigger={["mouseleave"]}
      addKeyboardHideShowTriggersByDefault={!showText}
    >
      <li
        id={id}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.BREADCRUMB_ITEM, id)}
        className={cx(styles.breadcrumbItemWrapper, className, {
          [styles.disabled]: disabled
        })}
      >
        <BreadcrumbContent
          ref={componentRef}
          isClickable={isClickable}
          link={link}
          onClick={onClick}
          text={text}
          icon={icon}
          isCurrent={isCurrent}
          disabled={disabled}
          showText={showText}
        />
      </li>
    </Tooltip>
  );
};

export default BreadcrumbItem;
