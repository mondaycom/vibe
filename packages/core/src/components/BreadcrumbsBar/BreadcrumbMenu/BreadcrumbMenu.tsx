import React, { forwardRef } from "react";
import MenuButton from "../../MenuButton/MenuButton";
import { Menu } from "../../Menu";
import { VibeComponentProps } from "../../../types";
import styles from "./BreadcrumbsMenu.module.scss";
import { ComponentDefaultTestId } from "../../../tests/constants";
import { getTestId } from "../../../tests/test-ids-utils";

export interface BreadcrumbMenuProps extends VibeComponentProps {
  children: React.ReactNode;
}
const BreadcrumbMenu = forwardRef(
  (
    { children, id, "data-testid": dataTestId, ...props }: BreadcrumbMenuProps,
    ref: React.ForwardedRef<HTMLLIElement>
  ) => {
    const menuButtonId = id || "breadcrumb-menu-button";
    const menuId = `${menuButtonId}-menu`;

    return (
      <li className={styles.breadcrumbMenuWrapper} {...props}>
        <MenuButton
          id={menuButtonId}
          size={"xxs"}
          closeMenuOnItemClick
          ariaLabel={"Hidden breadcrumbs menu"}
          data-testid={dataTestId || getTestId(ComponentDefaultTestId.BREADCRUMB_MENU, id)}
          ariaControls={menuId}
          ref={ref}
        >
          <Menu id={menuId} size={"medium"} focusItemIndexOnMount={0} ariaLabel="Expanded breadcrumbs menu">
            {children}
          </Menu>
        </MenuButton>
      </li>
    );
  }
);

export default BreadcrumbMenu;
