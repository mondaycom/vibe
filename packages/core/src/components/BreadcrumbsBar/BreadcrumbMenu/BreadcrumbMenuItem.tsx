import React, { forwardRef, ReactElement, Ref } from "react";
import MenuItem from "../../Menu/MenuItem/MenuItem";
import { MenuItemProps } from "../../Menu/MenuItem/MenuItem";
import { SubIcon, VibeComponent, VibeComponentProps } from "../../../types";
import { IconType } from "../../Icon/IconConstants";
import { ComponentDefaultTestId, getTestId } from "../../../tests/test-ids-utils";
import cx from "classnames";
import styles from "./BreadcrumbMenu.module.scss";

export interface BreadcrumbMenuItemProps extends VibeComponentProps {
  title?: string;
  icon?: SubIcon;
  iconType?: IconType;
  disabled?: boolean;
  onClick?: (event: React.MouseEvent) => void;
  active?: boolean;
}

export const BreadcrumbMenuItem: VibeComponent<BreadcrumbMenuItemProps, HTMLElement> = forwardRef(
  (
    {
      className,
      id,
      "data-testid": dataTestId,
      title = "",
      icon = null,
      iconType = IconType.SVG,
      disabled = false,
      onClick = undefined,
      active = false
    },
    ref: Ref<HTMLElement>
  ) => {
    const testId = dataTestId || getTestId(ComponentDefaultTestId.BREADCRUMB_MENU_ITEM, id);

    // We pass all props to MenuItem, as BreadcrumbMenuItem is just a contextual wrapper
    const menuItemProps: MenuItemProps = {
      className: cx(styles.breadcrumbMenuItem, className),
      id,
      "data-testid": testId,
      title,
      icon,
      iconType,
      disabled,
      onClick,
      selected: active // Map 'active' to 'selected' for MenuItem
    };

    // Pass the ref to the underlying MenuItem
    return <MenuItem {...menuItemProps} ref={ref} />;
  }
);
