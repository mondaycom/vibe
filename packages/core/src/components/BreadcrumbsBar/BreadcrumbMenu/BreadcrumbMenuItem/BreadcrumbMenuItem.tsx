import React, { forwardRef, ForwardRefExoticComponent, RefAttributes } from "react";
import { MenuItem, MenuItemProps } from "../../../Menu";
import { getTestId } from "../../../../tests/test-ids-utils";
import { ComponentDefaultTestId } from "../../../../tests/constants";

export interface BreadcrumbMenuItemProps extends MenuItemProps {
  /** Link to navigate to when item is clicked */
  link?: string;
}

interface BreadcrumbMenuItemComponent
  extends ForwardRefExoticComponent<BreadcrumbMenuItemProps & RefAttributes<unknown>> {
  isMenuChild?: boolean;
  isSelectable?: boolean;
}

const BreadcrumbMenuItem: BreadcrumbMenuItemComponent = forwardRef<unknown, BreadcrumbMenuItemProps>(
  (
    { id, "data-testid": dataTestId, link, onClick, ...rest }: BreadcrumbMenuItemProps,
    ref: React.ForwardedRef<HTMLLIElement>
  ) => {
    const handleClick = link
      ? (event: React.MouseEvent | React.KeyboardEvent) => {
          if (onClick) onClick(event);
          if (link) window.open(link);
        }
      : onClick;

    return (
      <MenuItem
        id={id}
        onClick={handleClick}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.BREADCRUMB_MENU_ITEM, id)}
        ref={ref}
        {...rest}
      />
    );
  }
);

BreadcrumbMenuItem.isMenuChild = true;
BreadcrumbMenuItem.isSelectable = true;

export default BreadcrumbMenuItem;
