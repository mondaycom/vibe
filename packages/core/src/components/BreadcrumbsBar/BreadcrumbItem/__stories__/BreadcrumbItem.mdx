import { Meta } from "@storybook/blocks";
import * as BreadcrumbItemStories from "./BreadcrumbItem.stories";

<Meta of={BreadcrumbItemStories} />

# BreadcrumbItem

- [Overview](#overview)
- [Props](#props)
- [Variants](#variants)
- [Feedback](#feedback)

## Overview

<Canvas of={BreadcrumbItemStories.Overview} />

### Import

```js
import { BreadcrumbItem } from "@vibe/core";
```

## Props

<PropsTable />

## Variants

### States

<Canvas of={BreadcrumbItemStories.States} />

### With icon

<Canvas of={BreadcrumbItemStories.WithIcon} />
