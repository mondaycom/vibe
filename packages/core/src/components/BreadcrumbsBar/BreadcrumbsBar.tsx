import cx from "classnames";
import React, { FC, ReactElement } from "react";
import { ComponentDefaultTestId, getTestId } from "../../tests/test-ids-utils";
import { NavigationChevronRight } from "@vibe/icons";
import { BreadcrumbsBarType as BreadcrumbsBarTypeEnum } from "./BreadcrumbsConstants";
import { BreadcrumbsBarType } from "./Breadcrumbs.types";
import { BreadcrumbItemProps } from "./BreadcrumbItem/BreadcrumbItem";
import { withStaticProps, VibeComponentProps } from "../../types";
import styles from "./BreadcrumbsBar.module.scss";

export interface BreadcrumbBarProps extends VibeComponentProps {
  /**
   * The type of the breadcrumb bar, determining if it is navigational or for indication only.
   */
  type: BreadcrumbsBarType;
  /**
   * The breadcrumb items displayed in the bar.
   */
  children: ReactElement<BreadcrumbItemProps> | ReactElement<BreadcrumbItemProps>[];
}

const BreadcrumbsBar: FC<BreadcrumbBarProps> & { types?: typeof BreadcrumbsBarTypeEnum } = ({
  className,
  children,
  type = "indication",
  id,
  "data-testid": dataTestId
}: BreadcrumbBarProps) => (
  <nav
    aria-label="Breadcrumb"
    className={cx(styles.breadcrumbsBar, className)}
    id={id}
    data-testid={dataTestId || getTestId(ComponentDefaultTestId.BREADCRUMBS_BAR, id)}
  >
    <ol>
      {children &&
        React.Children.map(children, (child, index) =>
          React.isValidElement(child)
            ? [
                index > 0 && <NavigationChevronRight className={styles.separatorIcon} size="14" aria-hidden="true" />,
                React.cloneElement(child, {
                  ...child?.props,
                  isClickable: type !== "indication"
                })
              ]
            : null
        )}
    </ol>
  </nav>
);

export default withStaticProps(BreadcrumbsBar, {
  types: BreadcrumbsBarTypeEnum
});
