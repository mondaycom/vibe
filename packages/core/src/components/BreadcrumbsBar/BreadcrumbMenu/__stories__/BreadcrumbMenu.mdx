import { Canvas, Meta } from "@storybook/blocks";
import * as BreadcrumbMenuStories from "./BreadcrumbMenu.stories";

<Meta of={BreadcrumbMenuStories} />

# BreadcrumbMenu

- [Overview](#overview)
- [Props](#props)
- [Variants](#variants)
- [Feedback](#feedback)

## Overview

<Canvas of={BreadcrumbMenuStories.Overview} />

### Import

```js
import { BreadcrumbMenu } from "@vibe/core";
```

## Props

<PropsTable />

## Variants

### States

<Canvas of={BreadcrumbMenuStories.States} />

### With Links

<Canvas of={BreadcrumbMenuStories.WithLinks} /> 