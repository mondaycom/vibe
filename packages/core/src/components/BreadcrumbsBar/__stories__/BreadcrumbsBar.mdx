import { Meta } from "@storybook/blocks";
import BreadcrumbsBar from "../BreadcrumbsBar";
import BreadcrumbItem from "../BreadcrumbItem/BreadcrumbItem.tsx";
import Avatar from "../../Avatar/Avatar";
import DialogContentContainer from "../../DialogContentContainer/DialogContentContainer";
import { Board, Folder, Group, Workspace } from "@vibe/icons";
import person3 from "./assets/person3.png";
import {
  BUTTON_GROUP,
  MULTI_STEP_INDICATOR,
  TABS
} from "../../../storybook/components/related-components/component-description-map";
import { TipCheckYourself } from "./BreadcrumbsBar.stories.helpers";
import * as BreadcrumbsBarStories from "./BreadcrumbsBar.stories";

<Meta of={BreadcrumbsBarStories} />

# BreadcrumbsBar

- [Overview](#overview)
- [Props](#props)
- [Usage](#usage)
- [Variants](#variants)
- [Do’s and don’ts](#dos-and-donts)
- [Use cases and examples](#use-cases-and-examples)
- [Related components](#related-components)
- [Feedback](#feedback)

## Overview

Breadcrumbs allow users to keep track and maintain awareness of their location as they navigate through pages, folders, files, etc.

<Canvas of={BreadcrumbsBarStories.Overview} />

### Import

```js
import { BreadcrumbsBar } from "@vibe/core";
```

## Props

<PropsTable />

## Usage

<UsageGuidelines
  guidelines={[
    "Breadcrumbs show users their current location relative to the information architecture and enable them to quickly move up to a parent level or previous step",
    "Effective in a hierarchy of more than two levels.",
    "Start with the highest level parent page and move deeper into the information architecture as the breadcrumb trail progresses.",
    "Use breadcrumbs when the user is most likely to have landed on the page from an external source.",
    "Place breadcrumbs at the top left corner of the screen, above the page title."
  ]}
/>

<TipCheckYourself />

## Variants

### Text only

<Canvas of={BreadcrumbsBarStories.TextOnly} />

### With icons

<Canvas of={BreadcrumbsBarStories.WithIcons} />

### With Breadcrumb Menu

<Canvas of={BreadcrumbsBarStories.WithBreadcrumbMenu} />

## Do’s and Don’ts

<ComponentRules
  rules={[
    {
      positive: {
        component: (
          <BreadcrumbsBar type="navigation">
            <BreadcrumbItem text="Workspace" icon={Workspace} isCurrent />
            <BreadcrumbItem text="Folder" icon={Folder} />
            <BreadcrumbItem text="Board" icon={Board} />
            <BreadcrumbItem text="Group" icon={Group} />
          </BreadcrumbsBar>
        ),
        description: "If there’s a need to insert an icon, use for all items."
      },
      negative: {
        component: (
          <BreadcrumbsBar type="navigation">
            <BreadcrumbItem text="Workspace" icon={Workspace} isCurrent />
            <BreadcrumbItem text="Folder" />
            <BreadcrumbItem text="Board" />
            <BreadcrumbItem text="Group" />
          </BreadcrumbsBar>
        ),
        description: "Don’t use icons if not applied to all steps."
      }
    },
    {
      positive: {
        component: (
          <DialogContentContainer className="monday-storybook-breadcrumbs-bar_inline-wrapper">
            <Avatar size="medium" src={person3} type="img" />
            <div className="monday-storybook-breadcrumbs-bar_column-wrapper">
              <span className="monday-storybook-breadcrumbs-bar_title">Rotem Dekel</span>
              <BreadcrumbsBar type="navigation">
                <BreadcrumbItem text="User research" icon={Board} />
                <BreadcrumbItem text="Video sessions" icon={Group} />
              </BreadcrumbsBar>
            </div>
          </DialogContentContainer>
        ),
        description: "Use breadcrumbs when there is more than two levels of hierarchy."
      },
      negative: {
        component: (
          <DialogContentContainer className="monday-storybook-breadcrumbs-bar_inline-wrapper">
            <Avatar size="medium" src={person3} type="img" />
            <div className="monday-storybook-breadcrumbs-bar_column-wrapper">
              <span className="monday-storybook-breadcrumbs-bar_title">Rotem Dekel</span>
              <BreadcrumbsBar type="navigation">
                <BreadcrumbItem text="User research" icon={Board} />
              </BreadcrumbsBar>
            </div>
          </DialogContentContainer>
        ),
        description: "Don’t use breadcrumbs when there is only one level of navigation or hierachy."
      }
    }
  ]}
/>

## Use cases and examples

### Navigatable breadcrumbs

Use when breadcrumbs are clickable and are used for information and navigation

<Canvas of={BreadcrumbsBarStories.NavigatableBreadcrumbs} />

## Related components

<RelatedComponents componentsNames={[BUTTON_GROUP, MULTI_STEP_INDICATOR, TABS]} />
