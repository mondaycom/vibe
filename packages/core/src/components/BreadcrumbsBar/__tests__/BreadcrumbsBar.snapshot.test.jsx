import { describe, it, expect } from "vitest";
import React from "react";
import renderer from "react-test-renderer";
import BreadcrumbsBar from "../BreadcrumbsBar";
import BreadcrumbItem from "../BreadcrumbItem/BreadcrumbItem";
import BreadcrumbMenu from "../BreadcrumbMenu/BreadcrumbMenu";
import BreadcrumbMenuItem from "../BreadcrumbMenu/BreadcrumbMenuItem/BreadcrumbMenuItem";

describe("BreadcrumbsBar renders correctly", () => {
  it("with empty props", () => {
    const tree = renderer.create(<BreadcrumbsBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with bar type", () => {
    Object.values(BreadcrumbsBar.types).forEach(type => {
      const tree = renderer.create(<BreadcrumbsBar type={type} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  it("with children", () => {
    const tree = renderer
      .create(
        <BreadcrumbsBar>
          <BreadcrumbItem text="Workspace" />
          <BreadcrumbItem text="Board" />
        </BreadcrumbsBar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("with breadcrumb menu", () => {
    const tree = renderer
      .create(
        <BreadcrumbsBar>
          <BreadcrumbItem text="Board" />
          <BreadcrumbMenu>
            <BreadcrumbMenuItem title="Item 1" />
            <BreadcrumbMenuItem title="Item 2" />
          </BreadcrumbMenu>
          <BreadcrumbItem text="My Item" isCurrent />
        </BreadcrumbsBar>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
