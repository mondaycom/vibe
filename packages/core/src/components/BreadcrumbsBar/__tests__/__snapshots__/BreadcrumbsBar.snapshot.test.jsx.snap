// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BreadcrumbsBar renders correctly > with bar type 1`] = `
<nav
  aria-label="Breadcrumb"
  className="breadcrumbsBar"
  data-testid="breadcrumbs-bar"
>
  <ol />
</nav>
`;

exports[`BreadcrumbsBar renders correctly > with bar type 2`] = `
<nav
  aria-label="Breadcrumb"
  className="breadcrumbsBar"
  data-testid="breadcrumbs-bar"
>
  <ol />
</nav>
`;

exports[`BreadcrumbsBar renders correctly > with breadcrumb menu 1`] = `
<nav
  aria-label="Breadcrumb"
  className="breadcrumbsBar"
  data-testid="breadcrumbs-bar"
>
  <ol>
    <li
      className="breadcrumbItemWrapper"
      data-testid="breadcrumb-item"
    >
      <span
        aria-disabled="true"
        className="breadcrumbContent notClickable"
        tabIndex={0}
      >
        <span
          className="breadcrumbText"
        >
          Board
        </span>
      </span>
    </li>
    <svg
      aria-hidden="true"
      className="separatorIcon"
      fill="currentColor"
      height="14"
      viewBox="0 0 20 20"
      width="14"
    >
      <path
        clipRule="evenodd"
        d="M13.5303 9.46967L13 10L13.5303 10.5303C13.8232 10.2374 13.8232 9.76256 13.5303 9.46967ZM11.9393 10L5.46967 16.4697C5.17678 16.7626 5.17678 17.2374 5.46967 17.5303C5.76256 17.8232 6.23744 17.8232 6.53033 17.5303L13.5303 10.5303L13 10L13.5303 9.46967L6.53033 2.46967C6.23744 2.17678 5.76256 2.17678 5.46967 2.46967C5.17678 2.76256 5.17678 3.23744 5.46967 3.53033L11.9393 10Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
    <li
      className="breadcrumbMenuWrapper"
      isClickable={false}
    >
      <button
        aria-controls="breadcrumb-menu-button-menu"
        aria-disabled={false}
        aria-expanded={false}
        aria-haspopup="true"
        aria-label="Hidden breadcrumbs menu"
        className="wrapper sizeXxs"
        data-testid="breadcrumb-menu"
        data-vibe="MenuButton"
        id="breadcrumb-menu-button"
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          height="16"
          role="img"
          viewBox="0 0 20 20"
          width="16"
        >
          <path
            d="M6 10.5C6 11.3284 5.32843 12 4.5 12 3.67157 12 3 11.3284 3 10.5 3 9.67157 3.67157 9 4.5 9 5.32843 9 6 9.67157 6 10.5zM11.8333 10.5C11.8333 11.3284 11.1618 12 10.3333 12 9.50492 12 8.83334 11.3284 8.83334 10.5 8.83334 9.67157 9.50492 9 10.3333 9 11.1618 9 11.8333 9.67157 11.8333 10.5zM17.6667 10.5C17.6667 11.3284 16.9951 12 16.1667 12 15.3383 12 14.6667 11.3284 14.6667 10.5 14.6667 9.67157 15.3383 9 16.1667 9 16.9951 9 17.6667 9.67157 17.6667 10.5z"
            fill="currentColor"
          />
        </svg>
      </button>
    </li>
    <svg
      aria-hidden="true"
      className="separatorIcon"
      fill="currentColor"
      height="14"
      viewBox="0 0 20 20"
      width="14"
    >
      <path
        clipRule="evenodd"
        d="M13.5303 9.46967L13 10L13.5303 10.5303C13.8232 10.2374 13.8232 9.76256 13.5303 9.46967ZM11.9393 10L5.46967 16.4697C5.17678 16.7626 5.17678 17.2374 5.46967 17.5303C5.76256 17.8232 6.23744 17.8232 6.53033 17.5303L13.5303 10.5303L13 10L13.5303 9.46967L6.53033 2.46967C6.23744 2.17678 5.76256 2.17678 5.46967 2.46967C5.17678 2.76256 5.17678 3.23744 5.46967 3.53033L11.9393 10Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
    <li
      className="breadcrumbItemWrapper"
      data-testid="breadcrumb-item"
    >
      <span
        aria-current="page"
        aria-disabled="true"
        className="breadcrumbContent current notClickable"
        tabIndex={0}
      >
        <span
          className="breadcrumbText"
        >
          My Item
        </span>
      </span>
    </li>
  </ol>
</nav>
`;

exports[`BreadcrumbsBar renders correctly > with children 1`] = `
<nav
  aria-label="Breadcrumb"
  className="breadcrumbsBar"
  data-testid="breadcrumbs-bar"
>
  <ol>
    <li
      className="breadcrumbItemWrapper"
      data-testid="breadcrumb-item"
    >
      <span
        aria-disabled="true"
        className="breadcrumbContent notClickable"
        tabIndex={0}
      >
        <span
          className="breadcrumbText"
        >
          Workspace
        </span>
      </span>
    </li>
    <svg
      aria-hidden="true"
      className="separatorIcon"
      fill="currentColor"
      height="14"
      viewBox="0 0 20 20"
      width="14"
    >
      <path
        clipRule="evenodd"
        d="M13.5303 9.46967L13 10L13.5303 10.5303C13.8232 10.2374 13.8232 9.76256 13.5303 9.46967ZM11.9393 10L5.46967 16.4697C5.17678 16.7626 5.17678 17.2374 5.46967 17.5303C5.76256 17.8232 6.23744 17.8232 6.53033 17.5303L13.5303 10.5303L13 10L13.5303 9.46967L6.53033 2.46967C6.23744 2.17678 5.76256 2.17678 5.46967 2.46967C5.17678 2.76256 5.17678 3.23744 5.46967 3.53033L11.9393 10Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
    <li
      className="breadcrumbItemWrapper"
      data-testid="breadcrumb-item"
    >
      <span
        aria-disabled="true"
        className="breadcrumbContent notClickable"
        tabIndex={0}
      >
        <span
          className="breadcrumbText"
        >
          Board
        </span>
      </span>
    </li>
  </ol>
</nav>
`;

exports[`BreadcrumbsBar renders correctly > with empty props 1`] = `
<nav
  aria-label="Breadcrumb"
  className="breadcrumbsBar"
  data-testid="breadcrumbs-bar"
>
  <ol />
</nav>
`;
